classy_vision/optim/adam.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Any, Dict, Tuple
+
+import torch.optim
+from classy_vision.generic.util import is_pos_float
+from classy_vision.optim.param_scheduler import (
+    ClassyParamScheduler,
+    build_param_scheduler,
+)
+
+from . import ClassyOptimizer, register_optimizer
+
+
+@register_optimizer("adam")
+class Adam(ClassyOptimizer):
+    def __init__(
+        self,
+        lr_scheduler: ClassyParamScheduler,
+        betas: Tuple[float, float],
+        eps: float,
+        weight_decay: float,
+        amsgrad: bool,
+    ) -> None:
+        super().__init__(lr_scheduler=lr_scheduler)
+
+        self.betas = betas
+        self.eps = eps
+        self.weight_decay = weight_decay
+        self.amsgrad = amsgrad
+
+    def init_pytorch_optimizer(self, model) -> None:
+        super().init_pytorch_optimizer(model)
+        self.optimizer = torch.optim.Adam(
+            self.param_groups_override,
+            lr=self.lr,
+            betas=self.betas,
+            eps=self.eps,
+            weight_decay=self.weight_decay,
+            amsgrad=self.amsgrad,
+        )
+
+    @classmethod
+    def from_config(cls, config: Dict[str, Any]) -> "Adam":
+        """Instantiates a Adam from a configuration.
+
+        Args:
+            config: A configuration for a Adam.
+                See :func:`__init__` for parameters expected in the config.
+
+        Returns:
+            A Adam instance.
+        """
+        # Default params
+        config.setdefault("eps", 1e-8)
+        config["amsgrad"] = config.get("amsgrad", False)
+
```

Comment:
We should also check if `betas` is a `list` and convert it to a `tuple` over here, since a JSON config can only have lists.